src/instrument.cpp

```diff
@@ -599,16 +827,34 @@ void JNICALL Instrument::ClassFileLoadHook(jvmtiEnv* jvmti, JNIEnv* jni,
     if (!_running) return;
 
     if (name == NULL || strcmp(name, _target_class) == 0) {
-        BytecodeRewriter rewriter(class_data, class_data_len, _target_class);
+        BytecodeRewriter rewriter(class_data, class_data_len, _target_class, _latency >= 0);
         rewriter.rewrite(new_class_data, new_class_data_len);
     }
 }
 
-void JNICALL Instrument::recordSample(JNIEnv* jni, jobject unused) {
+void JNICALL Instrument::recordEntry(JNIEnv* jni, jobject unused) {
     if (!_enabled) return;
 
+    if (_latency >= 0) {
+        _method_start_ns = OS::nanotime();
+        return;
+    }
     if (_interval <= 1 || ((atomicInc(_calls) + 1) % _interval) == 0) {
         ExecutionEvent event(TSC::ticks());
         Profiler::instance()->recordSample(NULL, _interval, INSTRUMENTED_METHOD, &event);
     }
 }
+
+void JNICALL Instrument::recordExit(JNIEnv* jni, jobject unused) {
+    if (!_enabled) return;
+
+    // TODO: does not account for recursive methods?
+    if (_method_start_ns != 0) {
+        u64 duration = OS::nanotime() - _method_start_ns;
+        _method_start_ns = 0;
+        if (duration >= _latency) {
+            ExecutionEvent event(TSC::ticks());
```

Comment:
The event `start_time` will be the method exit time, which may be confusing.

Reply:
e3eeebe4c3b26a3279669bc8fe42bda47e7842f5